### DISCUSSION TITLE
Multi-factor Authentication (MfA) Access Issue

### DISCUSSION THREAD

**Discussion by deh2025** [jgyates/genmon#1103] [General]
I enable MfA using Google Authenticator.  Not sure if it is a sync issue, but the random six-digit code is not recognize during the second part of the authentication process.  Is there a way to backdoor access Genmon via the Pi and deselect the MfA feature on the GUI?
Thanks!

**Comment by skipfire** [comment]
You can SSH in and edit the config file directly.  I believe MFA is time sensitive, so if your Pi is out of sync that could cause it to fail.  I believe `/etc/genmon/genmon.conf` is the file you want.

**Reply by jgyates** [reply]
yes in /etc/genmon/genmon.conf set usemfa=False to disable MFA

**Comment by deh2025** [comment]
I tried to SSH in, but when it prompts for the password, it allows no keyboard entries into the terminal emulator.  Very strange.

**Reply by skipfire** [reply]
Type the password and hit enter anyway, most systems don't show password entries even with masked characters as a security measure.

**Reply by skipfire** [reply]
Or if you have a keyboard and monitor connected you can do it through terminal on the Pi OS.

**Reply by jgyates** [reply]
are you using ssh on Mac or PuTTY on windows or some other ssh client? It could be that the terminal emulation on PuTTY is not set as the Pi is expecting? Have you ever used SSH on this pi before? Have you used SSH with this client before?

**Comment by deh2025** [comment]
So, you were absolutely correct, ssh accepted the password.  I had not thought about the masking - thanks!
Below is the output - not sure that it was of help.

Linux Genmon 6.1.21-v7+ #1642 SMP Mon Apr  3 17:20:52 BST 2023 armv7l

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Thu Apr 11 18:53:27 2024

**Reply by skipfire** [reply]
`sudo nano /etc/genmon/genmon.conf` to open the editor, then scroll down until you find `usemfa` and set it to false.  Then if you are on Windows you will press Ctrl+X to exit and (not sure about the right Ctrl key on Mac) it will prompt you to confirm you want to save it or not.  Then
`cd genmon` to switch to the genmon folder then
`./startgenmon.sh restart` to reboot genmon so that it reads the modified config.

All that is from memory as I'm on my phone, hopefully that is all correct.

**Comment by deh2025** [comment]
Actually, I am using ssh on Pi's browser rather than remoting via Windows or a Mac.

**Comment by deh2025** [comment]
Your memory was perfect!  I now have access to the GUI.  Much appreciated!!

**Comment by deh2025** [comment]
Thank you Jon. I appreciate the rescue!

I think I am finally ready for a dry run tomorrow.

One other unrelated question. Any idea to eliminate the following messages
that I receiving. It is a bit annoying, since I receive multiples.

Address not found
Your message wasn't delivered to ***@***.**** because the address
couldn't be found, or is unable to receive mail.

**Reply by jgyates** [reply]
Where are you seeing this message? A log file? in an email? 

Have you successfully sent a test email from the web interface settings page? 

First I would check the email addresses on the Notification page of the web interface and make sure that you have entered the email addresses correctly.

**Comment by deh2025** [comment]
It is being sent to my email.  Yes, the Genmon test messages have be successfully received.

Also, oddly I can only access the GUI locally from the Pi using the Chromium Browser. Entering the IP address of the Genmon times out on other devices.

**Reply by jgyates** [reply]
I would concentrate on the inability to access genmon on your local network first. These issues may be related. 

Can you give more details as it is not clear. You said " I can only access the GUI locally from the Pi using the Chromium Browser", how are you getting access to the pi, via ssh? Once on the pi can you perform a ping test to www.google.com? Open a console windows from the pi GUI and type

      ping www.google.com -c 4

If successfule it should look something like this:

     ping www.google.com -c 4

      PING www.google.com (64.233.185.105): 56 data bytes
      64 bytes from 64.233.185.105: icmp_seq=0 ttl=105 time=28.412 ms
      64 bytes from 64.233.185.105: icmp_seq=1 ttl=105 time=32.033 ms
      64 bytes from 64.233.185.105: icmp_seq=2 ttl=105 time=29.089 ms
      64 bytes from 64.233.185.105: icmp_seq=3 ttl=105 time=30.035 ms

If this fails then we should look at how your are getting and IP address for you pi, is is dynamically assigned or static. You may need to add your DNS server to the /etc/resolve.conf file. If you file does not have any entries then you can edit the file as described above when your were editing the /etc/genmon/genmon.conf file, then add this line:

      nameserver 8.8.8.8

This will make the pi use a google DNS server. Alternatively you can use your local routers IP address or your ISPs DNS server address.

**Reply by jgyates** [reply]
also, when you say "Entering the IP address of the Genmon times out on other devices", can you give more details? Are you entering the address in the browser, if yes, then what is the specific address you are using. If you use HTTPS then it would be something like:

https://192.168.1.10

if you are not using https then if you are using the default port the address would be like this:

http://192.168.1.10:8000

This assumes and IP address of 192.168.1.10, change that IP address to that of your pi.

**Comment by deh2025** [comment]
Looks like the test was successful.

--- www.google.com ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3004ms
rtt min/avg/max/mdev = 29.311/47.042/69.796/15.738 ms

I have a fairly extensive eero mesh network.  I placed it on my Guest network, since the primary's security features tends to interrupt some services, for example Mobile Link.

**Reply by jgyates** [reply]
can you ping from a computer on your non-guest network to the pi? and can you ping from the pi to a computer on your non-guest netowrk?

**Comment by deh2025** [comment]
I am using https://192.168.xx.xx in the Chrome browser on a Windows device.  I have not tried it on a Mac.  I have also verified the eero Pro that is currently assigned to its IP address.

**Reply by jgyates** [reply]
so if you put https://192.168.xx.xx/ in a browser on your primary network you can not connect to genmon, but you can ssh and ping the pi from your primary network? If that is the case, then either genmon is not running or your router is blocking access to web servers on your guest network.

Also, can you provide more details on this statment: "primary's security features tends to interrupt some services" are web sites being blocked? That could be happening to genmon as well.

Also, probalby would not hurt to validate your genmon web server settings on the Setting page under "Optional - Webserver Security Settings ". what are your settings. 

also, from a pi console, what is the output of this command:

     cat /etc/genmon/genserv.log

I am interested in the last 4 lines of that file.

**Reply by jgyates** [reply]
also, try this address:

    http://192.168.x.x:8000

Note this is http, not https

**Comment by deh2025** [comment]
Yes, actually I tried both on the primary and guest accounts.  I will give a try on the Mac, which is connected directly to the Gateway.

**Comment by deh2025** [comment]
The outcome on the Mac is the same.

This site can’t be reached

*192.168.xx.xx* took too long to respond.

Try:

   - Checking the connection
   - Checking the proxy and the firewall

ERR_CONNECTION_TIMED_OUT

**Comment by deh2025** [comment]
One point of good news. I sent a couple more test messages, which were
successfully received but not accompanied by an Address not found email.

**Comment by deh2025** [comment]
http://192.168.x.x:8000 - Output "This site can’t be reached"

cat /etc/genmon/genserv.log - No such file or directory  Should this script run a sub-directory?

**Reply by skipfire** [reply]
Ok, on guest it shouldn't work.  Guest networks should be isolated so that a guest cannot hack the main network and this is typically bidirectional.  

Just to be clear, you are replacing the xx.xx with the correct numbers, right?  Did you build this OS image yourself?

You want `cat /var/log/genserv.conf`.

**Reply by jgyates** [reply]
Sorry, it should be 

   cat /var/log/genserv.log

**Comment by deh2025** [comment]
Yes, using the xx.xx to hide the full IP address.
Thank you - cat /var/log/genserv.log - worked.  Here are the last few lined of the output.


2024-04-11 20:26:15,090 : Starting /home/Quail103/genmon/genserv.py, Port:443, Secure HTTP: True, SelfSignedCert: True, UseMFA:True
2024-04-11 20:26:45,293 : Admin Login
2024-04-11 20:27:41,943 : Admin Login
2024-04-11 20:29:27,809 : Starting /home/Quail103/genmon/genmon/genserv.py, Port:443, Secure HTTP: True, SelfSignedCert: True, UseMFA:True
2024-04-11 20:30:07,204 : Admin Login
2024-04-11 20:37:14,403 : Starting /home/Quail103/genmon/genserv.py, Port:443, Secure HTTP: True, SelfSignedCert: True, UseMFA:False
2024-04-11 20:37:32,460 : Admin Login
2024-04-11 20:41:34,173 : Script: /bin/bash /home/Quail103/genmon/startgenmon.sh restart -p 3 -c /etc/genmon/
2024-04-11 20:42:03,573 : Starting /home/Quail103/genmon/genserv.py, Port:443, Secure HTTP: True, SelfSignedCert: True, UseMFA:False
2024-04-11 20:42:23,456 : Admin Login

**Reply by skipfire** [reply]
Ah, try `https://192.168.xx.xx`

**Reply by skipfire** [reply]
You can also access by machine name depending on what name you set during setup.  Raspberry Pi Imager defaults to raspberrypi, so https://raspberrypi, since you have Secure HTTP turned on you won't need the :8000.

**Comment by deh2025** [comment]
I forgot to mention that I allowed the Pi and network reconcile the available IP address.  I found out the hard way not to hard code the IP address, too many potential conflicts.

**Reply by skipfire** [reply]
You can use a static IP, you just have to be sure it's outside the dynamic range that the router will assign.

**Comment by deh2025** [comment]
Ah, try https://192.168.xx.xx

That works only in Pi's browser.  It times out on other devices.  I have tried two Windows devices, an iMac, and my Android phone.

**Reply by skipfire** [reply]
And both the Pi and the devices you are trying from are on the primary wifi, none on the guest network?

**Comment by deh2025** [comment]
On the primary Windows device, I switched between both networks.  One the Mac and phone I was on the primary.

**Reply by skipfire** [reply]
Was the Pi on the primary?  You should not use guest for any of this as guest networks should not allow connectivity with devices on the primary network in either direction.

**Comment by deh2025** [comment]
I had another thought.  Could the issue be with the Pi's wifi configuration, which is independent of Genmon?

**Reply by skipfire** [reply]
Is the Pi connected to the guest or primary?  Past that, I don't believe there is anything on the Pi by default that would cause problems.

**Reply by deh2025** [reply]
The Pi is connected to the Guest network.

**Reply by skipfire** [reply]
As I've said a few times, it should not work if it is on Guest, guest networks are isolated so that the primary network is protected from them and the protection goes both ways.

**Reply by deh2025** [reply]
Understood, I will move it to primary.  Thanks

**Reply by deh2025** [reply]
Same outcome.  I had actually moved the Pi over to Guest yesterday thinking that the relaxed security might allow access.

**Reply by skipfire** [reply]
Did you update the IP address you are trying to to connect to?  Usually guest and primary have different ranges and it likely reassigned.

**Reply by deh2025** [reply]
I confirmed on the eero that the PI is now on primary.  I also rebooted my Windows laptop to clear the cache and tried again.  Unfortunately, no joy.

**Reply by skipfire** [reply]
Does your Pi have an Ethernet port?  If so, can you try that way?  That would ensure it is working aside from a WiFi setting.

**Reply by deh2025** [reply]
Yes, Pi3B+.  I will give it a try.  It is assigned to a different Pro than its currently assigned.  So this should be an interesting test.

Lastly, I went back an interrogated the eero Pro and discovered that the primary network assigned a new IP address.  I gave it a try with no success.

**Reply by deh2025** [reply]
That resulted in yet another newly assigned IP address, but again attempting access timed out.

**Reply by deh2025** [reply]
Okay, we are now up.  I moved everything to primary.  I have a couple spare Pros, so I can continue to test from here.  My plan configuration has the Pi remoted into the house.  Worst case I abandon the wifi option.  Thank you for your time and patience!
